src/Analyser/MutatingScope.php

```diff
@@ -1019,6 +1020,15 @@ private function resolveType(Expr $node): Type
 				return TypeCombinator::union(...$newTypes);
 			}
 
+			if ($type instanceof IntegerRangeType) {
+				$negativeRange = $this->resolveType(new Node\Expr\BinaryOp\Mul($node->expr, new LNumber(-1)));
+
+				if ( $negativeRange instanceof IntegerRangeType && ($negativeRange->getMin() === null || $negativeRange->getMax() === null)) {
```

Comment:
Actually I was not sure whether multiplication by -1 should have already returned the range as we need it and this if-case actually compensates a bug in the Mul logic?

Reply:
I don't know, you tell me :) Feel free to fix more bugs here.

Reply:
Is `-$b === $b * -1`?

Reply:
Seems like it. Here's a small script I made to generate equations for [Wolfram](https://www.wolframalpha.com/input/?i=%28interval%5B-%E2%88%9E%2C2%5D+*+-1%29+%3D+-%28interval%5B-%E2%88%9E%2C2%5D%29%3B): https://3v4l.org/AUHH7

Reply:
Fix in https://github.com/phpstan/phpstan-src/pull/670

Reply:
btw: the same is true for division: https://www.wolframalpha.com/input/?i=%28interval%5B-%E2%88%9E%2C%E2%88%9E%5D+%2F+-2%29+%3D+-%28interval%5B-%E2%88%9E%2C%E2%88%9E%5D%29%3B